src/routes/chart/get-panels.ts

```diff
@@ -0,0 +1,66 @@
+import Joi from 'joi';
+import { ControllerGeneratorOptionsWithClient } from '../../comms/controller';
+import { Chart, schema as chartSchema } from '../../models/chart';
+
+type HashId = string;
+
+// 'edgeHashIds', 'gridHashIds'
+const kindsHashIds = ['pinGroupHashIds', 'pinHashIds'] as const;
+type KindsHashIds = typeof kindsHashIds[number];
+
+// 'edges', 'grids'
+const kindsResources = ['pinGroups', 'pins'] as const;
+type KindsResources = typeof kindsResources[number];
+
+type Request = {
+  query: Record<KindsHashIds, Array<HashId>>;
+}
+
+type Panel = {
+  lastMode: 'manual' | 'automatic',
+  charts: {
+    automatic: Chart[],
+    manual: Chart[],
+  },
+};
+
+type Response = Record<KindsResources, Array<{
+  hashId: HashId,
+  panel: Panel,
+}>>;
+
+const charts = (apiVersion: number) => Joi.object().keys({
+  lastMode: Joi.string().valid('manual', 'automatic').example('automatic').required(),
+  charts: Joi.object().keys({
+    automatic: Joi.array().items(chartSchema(apiVersion)).required(),
+    manual: Joi.array().items(chartSchema(apiVersion)).required(),
+  }).required(),
+}).required();
+
+const controllerGeneratorOptions: ControllerGeneratorOptionsWithClient = {
+  method: 'get',
+  path: '/panels',
+  query: Joi.object().keys({
+    pinGroupHashIds: Joi.array().items(Joi.string()).max(30).required(),
+    pinHashIds: Joi.array().items(Joi.string()).max(30).required(),
+  }).required(),
+  right: { environment: 'READ' },
+  response: (apiVersion: number): Joi.ObjectSchema => Joi.object().keys({
+    pinGroups: Joi.array().items(Joi.object().keys({
+      hashId: Joi.string().required().example('dao97'),
+      panel: charts(apiVersion),
+    }).required()).required(),
+    pins: Joi.array().items(Joi.object().keys({
+      hashId: Joi.string().required().example('e13d57'),
+      panel: charts(apiVersion),
+    }).required()).required(),
+  }),
+  description: 'Get multiple chart panels from pinGroups, grids, edges and pins',
```

Comment:
Description here, comment line 7  and comment line 9 don't seem to match the implementation